src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,147 @@
+            return new ExitCommand();
+        } else if (fullCommand.trim().equals("undo")) {
+            return new UndoCommand();
+        } else {
+            //Splitting the full command into command and details.
+            String[] cmdAndDetails = fullCommand.split(" ", 2);
+            String cmd = cmdAndDetails[0];
+            Command command;
+            switch (cmd) {
+            case "delete":
+                command = createDeleteCommand(cmdAndDetails);
+                break;
+            case "done":
+                command = createDoneCommand(cmdAndDetails);
+                break;
+            case "todo":
+                command = createAddCommand(cmdAndDetails);
+                break;
+            case "deadline":
+                command = createAddCommand(cmdAndDetails);
+                break;
+            case "event":
+                command = createAddCommand(cmdAndDetails);
+                break;
+            case "find":
+                command = createFindCommand(cmdAndDetails);
+                break;
+            default:
+                throw new DukeException("OOPS!!! I'm sorry, but I don't know what that means :-(");
+            }
+            return command;
+        }
+
+    }
+
+    private static boolean isEmptyDescription(String description) {
+        return description.equals("");
+    }
+
+    /**
+     * Returns the Command object of DeleteCommand type.
+     *
+     * @param cmdAndDetails The array containing command and description of the command.
+     * @return The Command object of DeleteCommand type.
+     * @throws DukeException If the command has missing information or in wrong format.
+     */
+    private static Command createDeleteCommand(String[] cmdAndDetails) throws DukeException {
+        try {
+            String deleteTaskDescription = cmdAndDetails[1].trim();
+            if (isEmptyDescription(deleteTaskDescription)) {
+                throw new DukeException("OOPS!!! The description of delete cannot be empty.");
+            }
+            return new DeleteCommand(Integer.parseInt(deleteTaskDescription) - 1);
+        } catch (ArrayIndexOutOfBoundsException e) {
+            throw new DukeException("OOPS!!! The description of delete cannot be empty.");
+        } catch (NumberFormatException e) {
+            throw new DukeException("OOPS!!! The description of delete have to be a number.");
+        }
+    }
+
+    /**
+     * Returns the Command object of DoneCommand type.
+     *
+     * @param cmdAndDetails The array containing command and description of the command.
+     * @return The Command object of DoneCommand type.
+     * @throws DukeException If the command has missing information or in wrong format.
+     */
+    private static Command createDoneCommand(String[] cmdAndDetails) throws DukeException {
+        try {
+            String doneTaskDescription = cmdAndDetails[1].trim();
+            if (isEmptyDescription(doneTaskDescription)) {
```

Comment:
If this logic is same for all the commands, you can have it in a single place. Maybe before calling relevant methods.